### PR TITLE
CP-9399: K2 Alert

### PR DESCRIPTION
**Pull request by onghwan**
## Description

**Ticket: [CP-9399]** 

* Implement k2 alert for Alert use cases in the new design, especially for alerts having text inputs.
  * `showAlert()` function for simple alerts
  * `AlertWithTextInputs` component for the alerts with text inputs.

## Screenshots/Videos
| iOS | Android |
| --- | --- |
|  |  |

[CP-9399]: https://ava-labs.atlassian.net/browse/CP-9399?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/core-mobile/app/components/PasswordInput.tsx
Comment (onghwan) [inline packages/core-mobile/app/components/PasswordInput.tsx:1]: This file is no longer used.

File: packages/core-mobile/package.json [hunk 1]
```diff
@@ -65,7 +65,7 @@
     "@react-native-firebase/messaging": "21.0.0",
     "@react-native-google-signin/google-signin": "13.1.0",
     "@react-native-menu/menu": "1.1.3",
-    "@react-navigation/bottom-tabs": "6.6.1",
```
Comment (onghwan) [inline packages/core-mobile/package.json:68]: I had to downgrade this to match the version used in expo-router. If we don't, it raises an error when this package is used.

File: packages/k2-alpine/src/components/Alert/Alert.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
+import React, { useState } from 'react'
+import { KeyboardTypeOptions } from 'react-native'
+import Dialog from 'react-native-dialog'
+import { Alert as NativeAlert } from 'react-native'
+
+export const AlertWithTextInputs = ({
+  visible,
+  title,
+  description,
+  verticalButtons,
```
Comment (atn4z7) [inline packages/k2-alpine/src/components/Alert/Alert.tsx:10]: how does it look with verticalButtons?

File: packages/k2-alpine/src/components/Alert/Alert.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+            }
+          },
+          {
+            text: 'Delete',
+            style: 'destructive',
+            shouldDisable: (values: Record<string, string>) => {
+              return values.delete !== DELETE_TEXT
+            },
+            onPress: (values: Record<string, string>) => {
+              if (values.delete === DELETE_TEXT) {
```
Comment (atn4z7) [inline packages/k2-alpine/src/components/Alert/Alert.stories.tsx:81]: feel like this is not necessary. can't think of a case where we would need this.

**Review by atn4z7 — COMMENTED** [review]
looks great. left some minor comments.

File: packages/k2-alpine/src/components/Alert/Alert.tsx (hunk 2 above)
Comment (onghwan) [reply packages/k2-alpine/src/components/Alert/Alert.tsx:10]: * verticalButtons: false


* verticalButtons: true


verticalButtons:true will be useful when alert has more than two buttons.

File: packages/k2-alpine/src/components/Alert/Alert.tsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+      )}
+      {inputs &&
+        inputs.map(input => (
+          <Dialog.Input
+            key={input.key}
+            keyboardType={input.keyboardType}
+            secureTextEntry={input.secureTextEntry ?? false}
+            blurOnSubmit
+            onChangeText={(text: string) =>
+              setValues({ ...values, [input.key]: text })
```
[the remark this answers is not in the document]
Comment (onghwan) [reply packages/k2-alpine/src/components/Alert/Alert.tsx]: fixed.

**Review by atn4z7 — APPROVED** [review]

**Pull request merged**